blueocean-pipeline-editor/src/main/js/components/editor/EditorMain.jsx

```diff
@@ -75,8 +90,17 @@ export class EditorMain extends Component<DefaultProps, Props, State> {
         });
     }
 
+    componentDidMount() {
+        document.addEventListener('keydown', this.validateAfterTyping = debounce(e => {
+            if (!e.target.classList.contains('stage-name-edit')) {
+                pipelineValidator.validate();
+            }
+        }, 1000));
+    }
+
     componentWillUnmount() {
         pipelineStore.removeListener(this.pipelineUpdated);
+        document.removeEventListener('keydown', this.validateAfterTyping);
```

Comment:
This looks correct to me - although you might want to double check it works as expected. I think as long as you're not using `bind()` you are okay.

Reply:
yeah, this is fine, using the new syntax and storing a reference to the function, just above